cli/main.c

```diff
@@ -1433,6 +1433,108 @@ static int fw_toggle(int argc, char **argv)
 	return ret;
 }
 
+static int fw_redund_set(int argc, char **argv)
+{
+	const char *desc = "Set redundancy flag for a partition type";
+	int ret = 0;
+
+	const struct argconfig_choice types[] = {
+		{"KEY", SWITCHTEC_FW_TYPE_KEY,
+		 "Set the redundancy flag for the key manifest partitions"},
+		{"BL2", SWITCHTEC_FW_TYPE_BL2,
+		 "Set the redundancy flag for the bl2 partitions"},
+		{"CFG", SWITCHTEC_FW_TYPE_CFG,
+		 "Set the redundancy flag for the config partitions"},
+		{"IMG", SWITCHTEC_FW_TYPE_IMG,
+		 "Set the redundancy flag for the main firmware partitions"},
+		{}
+	};
+
+	static struct {
+		struct switchtec_dev *dev;
+		enum switchtec_fw_type type;
+	} cfg = {
+		.type = SWITCHTEC_FW_TYPE_UNKNOWN,
+	};
+
+	const struct argconfig_options opts[] = {
+		DEVICE_OPTION,
+		{"type", 't', "FW PARTITION TYPE", CFG_CHOICES, &cfg.type,
+		  required_argument,
+		 "firmware partition type to set", .choices=types},
+		{NULL}};
+
+	argconfig_parse(argc, argv, desc, opts, &cfg, sizeof(cfg));
+
+	if (switchtec_is_gen3(cfg.dev)) {
+		fprintf(stderr, "This command is not supported on Gen3 switch.\n");
+		return 1;
+	}
+
+	if (cfg.type == SWITCHTEC_FW_TYPE_UNKNOWN) {
+		argconfig_print_usage(opts);
+		fprintf(stderr, "The --type argument is required!\n");
+		return 2;
+	}
+
+	ret = switchtec_fw_setup_redundancy(cfg.dev, 1, cfg.type);
+	if (ret)
+		switchtec_perror("set fw redundancy");
+
+	return ret;
+}
+
+static int fw_redund_clear(int argc, char **argv)
```

Comment:
This function is nearly identical to the one above it. I'm sure one way or another you can make them the same function....

Reply:
Rebased and updated the CLI with adding only one command.